### DISCUSSION TITLE
moon culmination data problem _ Problème données Culminations lune

### DISCUSSION THREAD

**Discussion by ibaneoudo** [Stellarium/stellarium#2961] [Q&A]
Hello, With the astronomical calculations (F10 of the Stellarium interface) I asked to see the culminations of the moon over a given period. I then downloaded the data to excel. The problem is the following: the culmination times and the corresponding information given (height az; observer/moon distance, magnitude) are not always correct. There are in particular duplicates (two culminations for a single day and (then or before in the spreadsheet, everything is shifted. Is this a handling error on my part? a bug in the application? Thank you for your answer !

Bonjour, Avec les calculs astronomiques (F10 de l'interface Stellarium) j'ai demandé à voir les culminations de la lune sur une période données. J'ai ensuite télécharger les données sur excel. Le problème est le suivant : les heures de culminations et les informations données en correspondantes (hauteur az ; distance observateur/lune, magnitude) ne sont pas toujours bonnes. Il y a notamment des doublons (deux culminations pour une seule journée et (ensuite ou avant dans la feuille de calculs, tout est décalé. Est-ce une erreur de manipulation de ma part ? un bug de l'application ? Merci pour votre réponse !

**Comment by gzotti** [comment]
Yes, there is an occasional and annoying error with the Moon which we have not yet found. Two culminations per day are possible for the moon, though. Actually, two culminations for the same object happen every day in the minutes around midnight.

**Reply by ibaneoudo** [reply]
Thank you for your reply. Yes I understand the notion of two culminations of the moon in the same 24 hours. However, here, there are two culminations 1h00 or 2h00 apart. And the forecast hours do not resist the progressive approach (hour by hour for example on Stellarium) for its zenith passage due south observer. In fact, I do not know the method of calculation for the forecast of the culmination of the moon in Stellarium but we can see that this time changes a lot according to the "present" time indicated. It would be great to have in the data on the left when selecting a selest object the date corresponding to the day of the data presented. But also to have today's date with the culmination time. I do not know if such an addition of info is complex! If I can help, no problem (;-o). I use the ctrl shift C shortcut a lot to copy this famous informations from the selected object. Is there a shortcut or a way to copy the time and date of the selected data? (this is for python programming). Thanks!

**Reply by gzotti** [reply]
There is the AstroCalc dialog with ephemeris export to XLS.

**Reply by ibaneoudo** [reply]
thanks, yes, but the astrocalc take what position ? because with my local position, the xls are differents that info i find in stellarium !

**Reply by gzotti** [reply]
The XLS created on mine shows the same data as the list shown in AstroCalc. Culmination data are correct for the currently set location, although I see the occasional duplicate.

**Reply by ibaneoudo** [reply]
yes, same AstroCalc and the list, but very different when you verify on stellarium with the Wright date !!!

**Reply by gzotti** [reply]
It is not different on my PC. Care to show an example? What is different? Minutes? Hours? Random?

**Reply by ibaneoudo** [reply]
ok, i'm going to verify and show you an example

**Reply by ibaneoudo** [reply]
Do you have this too ? It may be an excel probleme too ? i don't known but my lists are wrong ! and too boring to to do corrections!
[CulminLune1711_2401.xlsx](https://github.com/Stellarium/stellarium/files/10372053/CulminLune1711_2401.xlsx)

**Reply by alex-w** [reply]
> Yes, there is an occasional and annoying error with the Moon which we have not yet found. Two culminations per day are possible for the moon, though. Actually, two culminations for the same object happen every day in the minutes around midnight.

We use one day step for compute the RTS for all objects, but the Moon have a fast movement and apparently we should use shorter time step.

**Reply by ibaneoudo** [reply]
Ok thank you, I understand a little better! It is for this reason that the results are not very precise when one is at a time far from the culmination of the following day. The problem being that we don't necessarily know the "day" date of the time mentioned in the selected object / data table. Which, depending on the data usage, can be very annoying !!!

**Comment by gzotti** [comment]
You seem to be somewhere near Brest, France, right? 
What version are you running?

**Reply by ibaneoudo** [reply]
yes, you right. I calculate from this position N 47° 49' 15.25" and O 4° 22' 27.23".
My stellarium's  version is 0.20.2.

**Reply by gzotti** [reply]
Sigh...

What to do when you have found a problem: UPDATE TO THE LATEST VERSION AND CHECK IF PROBLEM PERSISTS.

**Reply by ibaneoudo** [reply]
actually, sigh, updating me for the software would have been a good thing before meaning a malfunction. However, there is confusion between the astrocalc software and the user as to the definition of sunrise culmination and sunset. Depending on the local time, the data displayed by stellarium refer to the previous sunrise or the next sunrise. The astrocalc data materialize this confusion. So I come back to my first remark, the times of sunrise, culmination and sunset should mention the date of the day concerned. So everything I mention can't be clear because it's a confusing situation. The astrocalc of the culmination of 2023-01-20 12:04:03 does not know which days to choose for the risings (2023-01-20 08:15:15 or 2023-01-21 09:13:05) and bedtimes (2023-01-19 14:49:08 or 2023-01-20 15:54:03). As a user either, you don't always know if the times are the day, the day before or the next day because there are variations.
[CulminLune1711_2401_Stell.1.2.xlsx](https://github.com/Stellarium/stellarium/files/10380907/CulminLune1711_2401_Stell.1.2.xlsx)

**Reply by alex-w** [reply]
Version 1.2:

![stellarium-037](https://user-images.githubusercontent.com/88731/211511819-628d836a-c364-47b0-ba44-95bbe873f5ae.jpeg)

**Reply by ibaneoudo** [reply]
![Stell_RTS](https://user-images.githubusercontent.com/121743164/211515253-f26ba0bf-ce4d-47c8-a02b-264d3eaecc3f.jpg)

**Reply by ibaneoudo** [reply]
![Stell_RTS_zoom](https://user-images.githubusercontent.com/121743164/211516520-1882c5ea-1dfe-4482-aa2a-8177094cbd0d.jpg)
